auth/auth.go

```diff
@@ -135,6 +144,14 @@ func (auth *Authenticator) getPrincipal(docID string, factory func() Principal)
 				}
 				changed = true
 			}
+
+			if channelsPerUserThreshold := auth.ChannelsWarningThreshold(); channelsPerUserThreshold != nil {
+				channelCount := len(user.Channels())
```

Comment:
As discussed previously, we want this to take into account inherited channels too - those granted via roles. `user.Channels()` only fetches the channels associated directly with the user.

Reply:
Yeah, user.InheritedChannels() can be used instead of user.Channels(), which returns every channel the user has access to, including those inherited from Roles.